### PR TITLE
Improve DLP Processing

### PR DESCRIPTION
**Pull request by rdenarie**
Before this fix, when the number of element staying in queue is greater than the batch size, we are in infinite loop
This fix add a offset system for treating all the queue.

### CODE REVIEW

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java [hunk 1]
```diff
@@ -86,14 +87,20 @@ public synchronized void process() {
...
-      LOGGER.info("Dlp Operation Processor proceed {} queue elements by batches of {}", totalProcessed, BATCH_NUMBER_DEFAULT);
+      int offset = 0;
+      long total = dlpOperationDAO.count();
+      while (offset<total) {
+        LOGGER.debug("Process Bulk DLP operation (offset : {}, total {})", offset, total);
+        int stayingQueue=processBulk(offset);
+        int processedOperations=batchNumber-stayingQueue;
+        total=dlpOperationDAO.count();
+        offset+=stayingQueue;
+        totalProcessed=totalProcessed+processedOperations;
```
Comment (AzmiTouil) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java]: Format the code please !

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java (hunk 1 above)
Comment (rdenarie) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java]: Can you explain what is the problem in the format ?
Thanks

Comment (AzmiTouil) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java]:
```suggestion
      while (offset < total) {
        LOGGER.debug("Process Bulk DLP operation (offset : {}, total {})", offset, total);
        int stayingQueue = processBulk(offset);
        int processedOperations = batchNumber - stayingQueue;
        total = dlpOperationDAO.count();
        offset += stayingQueue;
        totalProcessed = totalProcessed + processedOperations;
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java (hunk 1 above)
Comment (rdenarie) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/processor/impl/DlpOperationProcessorImpl.java]: Done

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**